.github/workflows/pr.yaml

```diff
@@ -27,10 +27,29 @@ jobs:
           # Print this out for visibility.
           git rev-parse test-revision
 
-      - name: Ensure clang-format-18
+      - name: Install clang-format-19
         run: |
-          clang-format-18 --version
+          DISTRO_CODE='noble'
+
+          wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | sudo tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
+          sudo add-apt-repository -u "deb http://apt.llvm.org/${DISTRO_CODE}/ llvm-toolchain-${DISTRO_CODE}-19 main"
+
+          sudo apt-get install clang-format-19
+
+          clang-format-19 --version
 
       - name: Check formatting
         run: |
-          git clang-format-18 --diff ${GITHUB_BASE_REF} test-revision --verbose
+          git clang-format-19 --binary 'clang-format-19' \
+            --diff ${GITHUB_BASE_REF} test-revision --verbose
+
+      - name: Check formatting success
+        if: steps.run.outputs.status != '0'
+        run: |
+          cat <<EOF
+            =======
+            Has your style check failed? Go to sddf/ci/README to learn more
+            =======
+          EOF
+
+          exit "${{ steps.run.outputs.status }}"
```

Comment:
It needs to be in the same step as the diff output otherwise people won't see it.

Reply:
So the `Check formatting` step.